File: easy/23. Run Length Encoding/run_length_encoding.cpp
#include <string>

void numerificate(std::string &result, char c, int count) {
  if (count > 9) {
    while (count > 9) {
      result += std::to_string(9) + c;
      count -= 9;
    }
  }
  result += std::to_string(count) + c;
}

std::string runLengthEncoding(std::string str) {
  std::string result = "";
  int count = 1;
  for (int i = 1; i < str.length(); i++) {
    if (str[i] == str[i - 1]) {
      count++;
    } else {
      numerificate(result, str[i - 1], count);
  count = 1;
    }
  }
  numerificate(result, str[str.length() - 1], count);
  return result;
}

File: easy/23. Run Length Encoding/run_length_encoding.go
package main

import "strconv"

func numerificate(result string, c string, count int) string {
	if count > 9 {
		for count > 9 {
			result += "9" + c
			count -= 9
		}
	}
	result += strconv.Itoa(count) + c
	return result
}

func RunLengthEncoding(str string) string {
	result := ""
	count := 1
	for i := 1; i < len(str); i++ {
		if str[i] == str[i-1] {
			count += 1
		} else {
			result = numerificate(result, string(str[i-1]), count)
		count = 1
		}
	}
	result = numerificate(result, string(str[len(str)-1]), count)
	return result
}
